pkg/util/docker/dockerfile/builder/client.go

```diff
@@ -134,6 +149,23 @@ func (e *ClientExecutor) Build(r io.Reader, args map[string]string) error {
 			},
 		}
 		if mustStart {
+			// Transient mounts only make sense on images that will be running processes
+			if len(e.TransientMounts) > 0 {
+				volumeName, err := randSeq(imageSafeCharacters, 24)
+				if err != nil {
+					return err
+				}
+				v, err := e.Client.CreateVolume(docker.CreateVolumeOptions{Name: volumeName})
+				if err != nil {
+					return err
+				}
+				defer e.cleanupVolume(volumeName)
+				sharedMount = v.Mountpoint
+				opts.HostConfig = &docker.HostConfig{
+					Binds: []string{sharedMount + ":/tmp/__temporarymount"},
+				}
```

Comment:
sorry but i'm seriously struggling to follow what is going on here...  it seems like you're creating a docker volume, then setting up a bindmount for that (empty) volume.

Then you copy the source paths the user specified into that volume?  Except when you do the copies, you are copying to /tmp/_temporarymount, which i would have thought is the path inside the container, not the path on the host that represents the volume.  So i don't see how that copy actually works (but i haven't dug into what e.Copy is actually doing).

Then you bindmount all those copied directories, again seemingly mounting from the target path of the bindmount, not the source(host) path, so the container is bindmounting from within itself?

I would have expected something much more straightforward (setup the host paths, mount them into the container, then delete the host paths before committing the image(I guess))